reco_utils/evaluation/python_evaluation.py

```diff
@@ -582,10 +582,6 @@ def get_top_k_items(dataframe, col_user=DEFAULT_USER_COL, col_rating=DEFAULT_RAT
     Return:
         pd.DataFrame: DataFrame of top k items for each user.
     """
-    tmp = dataframe.copy()
-    tmp[col_rating] = tmp[col_rating].astype(float)
-    return (
-        tmp.groupby(col_user, as_index=False)
-        .apply(lambda x: x.nlargest(k, col_rating))
-        .reset_index()
-    )
+    return (dataframe.groupby(col_user, as_index=False)
+            .apply(lambda x: x.nlargest(k, col_rating))
+            .reset_index(drop=True))
```

Comment:
This changes the returned result, doesn't it? (removes the user info)

Reply:
The returned results will only have the user col and a rating col which has the top-k list. But the original dataframe should be untouched. Is that the desired output format?

Reply:
Ah sorry, I see. It drops the useless index.